### PR TITLE
Ammolathe: The beginning of the end.

### PR DESCRIPTION
**Pull request by DrSnips**
This introduces the ammolathe. Its an autolathe designed for ammo, guns, etc. The current sprite is a red autolathe <s>because I can't think of a good sprite idea for it</s> so it fits the sec theme. It's also locked out of science NORMALLY by being **design disk only** and will go in the warden's room somewhere. 
**The mapping will come in a hotfix right after this since mapping makes mustard gas for some reason. You can scroll down to see where the planned locations will be.**
It can technically be unlocked with NT experimental 5 research but that's only obtained via admins. This is a workaround for technical reasons in research code.

Currently it can make the following list of items:
- Guns: vector, NT glock.

- Ammo: Most shotgun ammo, all 3 types of .380.

- Mags: Every mag type but special ones like the lawgiver's.

- Misc: All speedloaders and vector receiver.

- Contraband/hacked: sec boot knife, vials, and syringes. Technical ammo.

>Where's the everything else?

I want to do things in waves to discuss the content in bite sized pieces rather than dealing with 100 balance issues at once.

**Unatomic changes alert:** 1)Fabs now have an animation for inserting sheets of phazon. 2)Removes mags from lockboxed guns to fix issues they had with accepting mags.(The lathes can make their respective mags cheaply so it's not a huge issue IMO.)
:cl:
 * rscadd: Added the ammolathe. It can print ammo, guns, and gun accessories like speed loaders.

### CODE REVIEW

**Comment by DrSnips** [comment]
Inb4 "This PR SUX!!!1!11"

**Comment by Irradiation** [comment]
This PR SUCCS

**Comment by MadmanMartian** [comment]
Pewyar shucks

**Comment by DrSnips** [comment]
I updated the OP with a lot of the changes done so far. Feel free to point out any issues with the item list and whatnot. I plan to put the lathe in the pre-armory area and the design disk in the warden's room.
I'm still looking for suggestions on a better lathe sprite but a red themed autolathe is good for now.

**Comment by jknpj** [comment]
>can make vectors

wew

**Comment by DrSnips** [comment]
I'm open to adding/removing anything if it's too much for wave 0. I threw it in there since I think it's on par with dual wielded glocks and I just personally like the gun.

**Comment by Blithering** [comment]
I don't understand the reasoning for a hacked ammo printer making blindfolds and the like, wouldn't silencers and modkits be more thematic?

**Comment by DrSnips** [comment]
It was a request in the discord and it wasn't too overwhelming or controversial for wave 0 content.

**Comment by ghost** [comment]
Can we not do this big song and dance again and don't give me that it'll be fine shit we all know the slopes lubed.

File: code/game/machinery/ammolathe.dm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	icon_state = "ammolathe"
+	icon_state_open = "ammolathe_t"
+	nano_file = "ammolathe.tmpl"
+	default_mat_overlays = TRUE
+
+	start_end_anims = 1
+
+	build_time = 0.5
+
+	allowed_materials = 0
```
Comment (MadmanMartian) [inline code/game/machinery/ammolathe.dm]: What would this do?

File: code/game/machinery/ammolathe.dm [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		new /obj/item/ammo_storage/magazine/a762/empty(), \
+		new /obj/item/ammo_storage/magazine/a12ga/empty(), \
+		),
+		"Misc_Other"=list(
+		new /obj/item/ammo_storage/speedloader/c38/empty(), \
+		new /obj/item/ammo_storage/speedloader/a357/empty(), \
+		new /obj/item/ammo_storage/speedloader/a762x55/empty(), \
+		new /obj/item/ammo_storage/speedloader/shotgun(), \
+		),
+		"Hidden_Items" = list(
```
Comment (MadmanMartian) [inline code/game/machinery/ammolathe.dm:53]: These suck and turn it from ammolathe to seclathe.

Comment (DrSnips) [reply code/game/machinery/ammolathe.dm:53]: Alright I can remove the stuff from the contraband list since a few people have had an issue with it. Any suggestions on what to put in there that doesn't overlap with the above. This means stuff that's not box/single ammo, a gun, etc. I was thinking something like syringes and vials since those are technically ammo for lesser known arms.

File: code/game/machinery/ammolathe.dm (hunk 1 above)
Comment (DrSnips) [reply code/game/machinery/ammolathe.dm]: allowed_materials is a var for what the name implies. The autolathe limits it to glass and metal but setting it to false/0/etc removes restrictions inherited from the autolathe to allow all mat types.

**Comment by DrSnips** [comment]
@SarahJohnson489 Thanks for your opinion. If you have any actual direct concerns please say so rather than a general "This PR SUX!"

File: code/game/machinery/ammolathe.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/game/machinery/ammolathe.dm]: Ooh. I like it

**Comment by MadmanMartian** [comment]
Check if it works with the bluespace bins

**Comment by DrSnips** [comment]
Alright I will add it to my list of things to do. If it worked with the autolathe then it should work with this but I will make sure.

**Comment by DrSnips** [comment]
I tested bluespacebins and they work as normal.

**Comment by DrSnips** [comment]
Looks like something made mustard gas. Fucking m*ppers.

**Comment by DrSnips** [comment]
Mapping gave me cancer and bloody lungs and this PR is in a done state. I [likely someone else] can map the changes in after so the maps don't make mustard and cancer gas.

**Comment by DrSnips** [comment]
all maps ammolathe
Here's a quick image on where I plan to put things after this. Green boxes to highlight them and yellow arrows indicate if something was pushed to make room for the lathe like implant boxes being moved to the same spot as all of the other implants.

**Comment by ghost** [comment]
How about the some 70 duplicate PR's and issues related to the ammolathe nuclear option, gun erosion creep, material theft, power abuse, incompetence, understaffing, and so. Don't play this tee hee I don't get what you mean :^) shit like we don't both know where it's going.

**Comment by DrSnips** [comment]
Feel free to present direct concerns and I'm open to discussing them.
-Nuclear option
What? I don't recall anything having to deal with nuclear stuff. If you mean atomic additions I think that's a safer way to add things so you can vote yes and no to certain things.
-Gun erosion creep
Again I don't remember this ever being said anywhere. I'm honestly tempted to call you schizophrenic because you're hearing and seeing things no one else sees.
-Material theft
A valid concern. Sec players might take stuff from sci/cargo by force but miners might adjust their drop offs to include sec as well to provide materials to them and in exchange for guns to use on the roid as a win win.
-Power abuse
It's true that this makes groups of people more or less powerful IF they have materials. I intend to leave the design disk and lathe fairly accessable to antags and whatnot so they have a chance to take it for themself.
-incompetence, understaffing, and etc.
I don't see the connection here.

**Comment by ddolll** [comment]
This PR sucks

- Design disk only for some reason

- Adds crafting to sec (the department that's not supposed to have it)

- Uninteresting and unnecessarily overlaps with existing systems

**Comment by DrSnips** [comment]
The design disk part is so people with a circuit printer (namely sci and mechanics) can't just make it without sec permission. I like this solution better than a lockbox circuit.
I wouldn't quite call it crafting since is simple single button pushing. I do really want this for what the name says, ammo, since the only way to get ammo right now is by cargo or sci which is limited to only a few kinds of ammo.
I don't think redundancy is a bad thing here. You complain that you don't want actual crafting which can be interesting but then also criticize it for being uninteresting. You seem to have a conflicting stance if you want something interesting or not.

**Comment by ghost** [comment]
Okay so you weren't here and missed the entire shitflinging over guns way back. 

tl;dr there was a run made on sci guns and the ammolathe was the nuclear option because it seems reasonable but it's a deliberate trojan horse cause once it existed it was only a matter of time before the ones were left were sucked into it. This isn't hyperbole. This isn't fearmongering. This codebase is filled with 'well bad thing was done so worse thing is okay too' snowballing and trying over and over until one finally sticks tumors. Putting in the ammolathe is tearing off the hard fought scab by putting in place an existing system anyone only has to copy/paste into until one flies under the radar or there's enough waah I couldn't just walk into a department and kill everyone there was stopped code he out to stick. Then it's a much, much higher hill to get rid of once it's in.

The erosion played out in the 70 duplicate shitflinging pr's and issues over guns as it kept getting worse.

It's uncommon for mining to share. They either went with mechanics cause they trusted them or they went with science. It took bluespace beakers completely sidestepping any voluntary actions to get around mineral distribution. Security has zero interest in conserving diamonds for bags to stop a loose not that anyone even tries with their bags anymore but the problem here now is security is not only an actor but a major participant in the system. It was one thing to end up binning some well-meaning shit for brains that wouldn't stop stuffing bluespace bins in your personal fabs that held the stuff you mined for whatever you wanted to do or binning them when they try to steal your toolbox or backpack of minerals cause it's the only way to protect them. It's even worse if now it's someone from sec and the only way to stop this shit is to cut their head off and destroy it which will only spark a war. You're banned or robbed and admins don't do fucking shit either way and that's shit.

Connection's in the 70 duplicate shitflinging pr's and issues over guns. Guns will get sucked out of science if this goes through it's only a matter of time and with it it'll be a rerun of the headaches raised in those where it reduces the agency of the crew when things inevitably are shit.

**Comment by DrSnips** [comment]
I can't deny that I'm not as aware of our issues in the past (pre-2016) but I'd like to point out that players /coders have cycled in and out since then and current day isn't the same as whenever this all happened.
The last sentences in your first paragraph makes no sense.
I remember the broken contents from lockboxs debate. Is that what you meant by gun erosion?
The materials thing seems to be an personal issue. I can't fix how a player reacts to someone using the bluespacebin pool of mats. I wouldn't be offended in the slightest and I encourage it but then there's people with your presented stance where they would rather be banned than let a single sheet be used by anyone but them self like a cunt. Again I can't do anything about anger issues.
Guns getting removed from the protolathe is not even on the table right now and nor did I say anything about the protolathe getting or losing designs in the future besides the ammolathe not being a normal research design for the printer. Bother the people who make those PRs in the future because this currently has nothing to do with it.

**Comment by Exxion** [comment]
1. There's no reason this would necessarily result in guns being moved out of science
2. That isn't even a bad thing

**Comment by MadmanMartian** [comment]
Wouldn't moving some of the basic guns out of science be a good thing?

What's so experimental about an SMG from XCOM?

**Comment by DrSnips** [comment]
I'd rather not touch on the protolathe save copying some designs from the proto to the ammo. Currently the 9mm and 12mm mags and stun shells have overlap.

**Comment by MadmanMartian** [comment]
My suggestion: Have it purely as an ammolathe, no guns inside it at all.

**Comment by DrSnips** [comment]
I'd be fine with removing them but I'd like to hear it from a few other people as well since the people who voted for this voted for it while it had guns and ammo in mind.

File: code/game/machinery/ammolathe.dm (hunk 2 above)
Comment (DrSnips) [reply code/game/machinery/ammolathe.dm:53]: Item list changed but this didn't close.

**Comment by SonixApache** [comment]
i'll go with my initial suggestions:
 - menu to configure ammo types
 - place gun/magazine inside to analyze (for xenoarch?) and show available ammo types
 - recharge gun/magazine

On one hand, I think being able to create guns is fine, since there's plenty of other ways to create guns already.
On the other hand, there's less need for departmental interaction for firearms procurement.

**Comment by DrSnips** [comment]
For how the machine works I don't want to over complicate things that don't need it. Press button, receive bullets.
Why would I need to insert a mag/gun to "analyze" and other fluff shit for bullets if I know the glock is just going to use .380 and the shotgun is going to to need 12gauge?
You would still need a hand from cargo or someone who can mine mats like silver/gold/etc. to access some stuff. It could give traders an option if sci or whoever is being greedy with mats.
I personally don't care about forcing departmental interaction and I'm not going to force players to play a certain way because that's BIG gay.

**Comment by ghost** [comment]
Jesus it really does never end. 

Gun erosion is people continually trying to erode guns slicing away at their effectiveness, availability, doing shit like this where it puts in the framework for any copy paster to keep making pr's until one sticks.

It's not a personal issue. Sec doesn't have any reason to not empty the bins for whatever they want and if you think it's being a cunt to not want to be unable to do whatever you joined the round and mined to do because of some sociopathic asshole without having to go mine again you should fucking kill yourself instead of calling anything cunty.

It doesn't matter what you intended with the ammolathe, if they could have been fucked to actually make it back then all the guns would have slowly been sucked into it by now and they will when this goes through as people increasingly disconnected from the game see no problem with the move or even as a good thing. It has to be stopped here so the erosion doesn't start up all over again.


Exxion, you say "That isn't even a bad thing" but that's exactly how it's going to be pitched when it starts happening. It's a bad thing cause cargo is still a hard id/silicon lock and sec is still a DMZ you can get fucked for going in whereas science has mostly become accepted as a public onahole for whatever people want/need for their role and not getting boned by yet another ebin lowpop murderbone and chill for 2 hours.

**Comment by ddolll** [comment]
>Guns getting removed from the protolathe is not even on the table right now and nor did I say anything about the protolathe getting or losing designs in the future besides the ammolathe not being a normal research design for the printer.

It's a pretty huge fucking deal that you're literally taking away science's ability to access a section of crafting/printing. Any designs which are added to this and not to the autolathe/protolathe are literally designs being removed from the autolathe/protolathe which belong in there.

As an aside, the whole point of R&D being able to make machines was to be able to build anything from the ground up. Taking machines out of R&D directly works against the 'everything is constructable/deconstructable' ideal that's been worked on so much.

Anyway to reiterate the points I made in the discord because OP didn't voice a disagreement with any of them:

- When the protolathe was added, it was justified because it represented a new barrier of entry to put things behind to prevent the stronger items being added from overwhelming the (at the time, fast paced) game. The ammolathe doesn't represent a new process that has to be completed to lock things off in the way that the mining/research system does (it uses the same system even), so the actual machine itself is redundant. Anyone with all access or who isn't playing nice can still just make a complete R&D setup, except it needlessly has one more machine in it. The only difference here is that you've associated this new machine deeply with security, who have no positive (material/research generating) feedback within the system and who can only detract from the experience of the existing users of the R&D system, as SJ has outlined.

- Putting an R&D system directly into sec causes sec to get more powerful as the round goes on / as research and materials become available. This breaks antag/sec and maint/sec power relations because at present the R&D systems are tied to departments which are neutral to security; in the event of antagonist activity, whether the holders of the tools produced side with the station or the antagonists determines who gains an advantage from the research/mining having been done. What this means is that for things like cult or rev etc. the pre-R&D meta of 'whoever secures cargo can produce weapons' expanded to 'whoever secures cargo and active R&D can produce weapons'. With this additional, security will be able to scale in power regardless of whether they have control of or support from other departments, so long as they get materials (the process by which sec, as a department who do nothing to produce materials, obtain materials, is again fucking awful for round health). For solo antags, the station/antag power balance has been determined by whether the tools produced by R&D are in station or antag hands. With the ammolathe, the station (actually security) will always at least remain on par and usually gain an advantage from active R&D. Furthermore, this breaks maint/security balance because security will now (always) scale with research but maint will not.

- Autolathe has always been supposed to have every ammo type. It always had revolver (and shotgun) ammo back when revolvers and shotguns were the only bullet guns, and it's not as if there are ammo types you'd add to station availability classified at a higher level than 'traitor item' like revolver ammo was. There's no reason to cut gun distribution up. If you want guns to be accessible through R&D and it won't gunk the game just add the guns to R&D.

**Comment by DrSnips** [comment]
We are clearly never going to reach an end to this so I will just leave it here and let other people make their own decisions based on what we have said so far.

>Any designs which are added to this and not to the autolathe/protolathe are literally designs being removed from the autolathe/protolathe

I just want to address this hyper retarded leap in logic.
1.Where's the rule that because something is in one lathe means it MUST be reflected in another lathe?
2.You literally can't remove something from somewhere if it was never there in the first place.
Feel free to post your own responses so you can't say I twisted your words somehow.

**Comment by ddolll** [comment]
>I just want to address this hyper retarded leap in logic.
1.Where's the rule that because something is in one lathe means it MUST be reflected in another lathe?
2.You literally can't remove something from somewhere if it was never there in the first place.
Feel free to post your own responses so you can't say I twisted your words somehow.

You literally asked about this on the discord yet still felt the need to shitpost, but:
1. There is no such rule, that's the point. There also aren't 'lathes'; there is the protolathe and the autolathe. Outside of stock parts, these do not overlap. This is because the protolathe represents a new barrier of entry which was higher than that of the autolathe.
2. Any gun etc. which would get added to this lathe and not the protolathe is something which has been decided was okay to make available to the station via R&D, which has been added, and is not in the protolathe. If the ammolathe didn't exist, it would have been added to the protolathe where it belongs. This is removing it from the protolathe, because the thing otherwise would have been in the protolathe.

**Comment by SonixApache** [comment]
@DrSnips guns will use the same caliber, sure, but loading different types of bullets into guns/magazines (lethals, stuns, blanks, etc).

File: code/game/machinery/ammolathe.dm (hunk 2 above)
Comment (MadmanMartian) [reply code/game/machinery/ammolathe.dm:53]: I still spy that tactiknife

Comment (DrSnips) [reply code/game/machinery/ammolathe.dm:53]: I thought it would be fitting enough for a hidden option. I'm for leaving it in there but if some other people want it gone then I will cave.

Comment (DrSnips) [reply code/game/machinery/ammolathe.dm:53]: The knife gun apparently doesn't exist [yet] so they have been removed.

**Review by MadmanMartian — APPROVED** [review]
Not a fan that it makes guns, especially a gun compatible with pretty much any ammunition type. Thought this was going to be strictly an ammunition lathe.

File: code/modules/research/designs/weapons.dm [hunk 3]
```diff
@@ -257,3 +257,138 @@
 	materials = list(MAT_IRON = 12000)
 	category = "Weapons"
 	build_path = /obj/item/weapon/storage/pneumatic
+
+//Ammolathe designs. They lack research values so they can't be researched.
+//Weapons
+/obj/item/weapon/gun/projectile/sec
```
Comment (MadmanMartian) [inline code/modules/research/designs/weapons.dm]: ??

File: code/modules/research/designs/weapons.dm [hunk 4]
```diff
@@ -257,3 +257,138 @@
...
+	id = "glock"
+	build_type = AMMOLATHE
+	materials = list(MAT_IRON = 10000, MAT_GLASS = 10000)
+	build_path = /obj/item/weapon/gun/projectile/sec/lockbox
+	locked = TRUE
+	req_lock_access = list(access_armory, access_weapons)
+
+/datum/design/vector
+	name = "Vector"
+	desc = "A gun."
```
Comment (SonixApache) [inline code/modules/research/designs/weapons.dm]: 🤔

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 5]
```diff
@@ -166,6 +166,9 @@
 	..()
 	icon_state = "secglockfancy[chambered ? "" : "-e"][silenced ? "-s" : ""][stored_magazine ? "" : "-m"]"
 
+/obj/item/weapon/gun/projectile/sec/lockbox
+	mag_type = "/obj/item/ammo_storage/magazine/m380auto/empty"
```
Comment (SonixApache) [inline code/modules/projectiles/guns/projectile/pistol.dm]: did you check if this can still be loaded with the alternate magazines?

File: code/modules/research/designs/weapons.dm (hunk 4 above)
Comment (DrSnips) [reply code/modules/research/designs/weapons.dm]: I'm retarded and forgot to save/push changes one time it seems.

File: code/modules/projectiles/guns/projectile/pistol.dm (hunk 5 above)
Comment (DrSnips) [reply code/modules/projectiles/guns/projectile/pistol.dm]: No I have not. Someone else or myself has to fix that issue. I have an idea in mind but that's for another PR. I can remove the guns for now I guess till said fix or leave them in.

Comment (DrSnips) [reply code/modules/projectiles/guns/projectile/pistol.dm]: Just tested it. They cannot use the other ones. I can work on a fix for it today but not in this PR.

File: code/modules/research/designs/weapons.dm (hunk 3 above)
Comment (DrSnips) [reply code/modules/research/designs/weapons.dm]: Fixed.

File: code/modules/research/designs/weapons.dm (hunk 4 above)
Comment (DrSnips) [reply code/modules/research/designs/weapons.dm]: Fixed.

**Comment by DrSnips** [comment]
<s>Besides the lockbox issue</s> this PR is ready to go. The guns will stay for now until more ammo comes along that raises concerns.
The mapping is planned for a hotfix shortly after this PR because the maps have changed since this PR was made and mapping makes cancer gas for me. Merge this PR under the assumption that the map changes are a part of this current PR and if you have issues with that you can hold off on merging this.

File: code/modules/projectiles/guns/projectile/pistol.dm (hunk 5 above)
Comment (MadmanMartian) [reply code/modules/projectiles/guns/projectile/pistol.dm]: might as well fix it now

Comment (DrSnips) [reply code/modules/projectiles/guns/projectile/pistol.dm]: It would require making all the mags change in a new() for every single gun that should start empty. Are you ok with this level of unatomic?

Comment (SonixApache) [reply code/modules/projectiles/guns/projectile/pistol.dm]: IMO; no. We've discussed how to best solve this before, if you remember, and I think that's material for a very big and very seperate PR.
Better option for this possibly be to just ditch the options for now.

Comment (DrSnips) [reply code/modules/projectiles/guns/projectile/pistol.dm]: Issue resolved.

**Review by MadmanMartian — APPROVED** [review]
didn't need to be all of them, but 🤷‍♂️ whatever works

**Comment by DrSnips** [comment]
Just making it all uniform.
